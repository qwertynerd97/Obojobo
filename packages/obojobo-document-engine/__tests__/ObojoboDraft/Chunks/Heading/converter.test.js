jest.mock('src/scripts/oboeditor/util/text-util')

import Converter from 'ObojoboDraft/Chunks/Heading/converter'

describe('Heading Converter', () => {
	test('slateToObo converts a Slate node to an OboNode with content', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			data: {
				get: () => ({})
			},
			text: 'mockText',
			nodes: [
				{
					leaves: [
						{
							text: 'mockText',
							marks: [
								{
									type: 'b',
									data: {}
								}
							]
						}
					]
				}
			]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node with a caption', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: {
				textGroup: [
					{
						text: { value: 'mockText' }
					},
					{
						text: { value: 'mockText' },
						data: { align: 'left' }
					}
				]
			}
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})
})
