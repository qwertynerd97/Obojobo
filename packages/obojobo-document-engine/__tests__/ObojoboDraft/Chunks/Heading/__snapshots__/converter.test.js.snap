// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Converter oboToSlate converts an OboNode to a Slate node with a caption 1`] = `
Object {
  "data": Object {
    "content": Object {
      "align": "left",
      "level": undefined,
    },
  },
  "key": "mockKey",
  "nodes": Array [
    Object {
      "leaves": undefined,
      "object": "text",
    },
    Object {
      "leaves": undefined,
      "object": "text",
    },
  ],
  "object": "block",
  "type": "mockType",
}
`;

exports[`Heading Converter slateToObo converts a Slate node to an OboNode with content 1`] = `
Object {
  "children": Array [],
  "content": Object {
    "headingLevel": undefined,
    "textGroup": Array [
      Object {
        "data": Object {
          "align": undefined,
        },
        "text": Object {
          "styleList": Array [],
          "value": "mockText",
        },
      },
    ],
  },
  "id": "mockKey",
  "type": "mockType",
}
`;
